v-next/hardhat/src/types/hooks.ts

```diff
@@ -321,8 +308,11 @@ export interface HookManager {
   ): Promise<Awaited<Return<HookT>>>;
 
   /**
-   * Runs all the handlers for a hook in the same order that `getHandlers`
-   * returns them.
+   * Runs all the handlers for a hook in the following priority order:
+   *  - Dynamically registered handlers come first, in the reverse order they
+   *   were registered.
```

Comment:
There is an argument that the registered handlers should also not be reversed. But that was a step further than was needed to fix the underlying bug.

Are we happy that for sequential runs the ordering of registered handlers is done in the same was as for the handler chain?

Reply:
As susggested in another comment, I think this should be the reverse of the other one.